pyprotista/parsers/ident/comet_2020_01_4_parser.py

```diff
@@ -149,25 +197,23 @@ def _map_mods_and_sequences(self):
                 .agg(";".join, axis=1)
                 .str.rstrip(";")
             )
-
-        modification_mass_map = {
-            sm.attrib["massDelta"]: sm.find(".//{*}cvParam[@cvRef='UNIMOD']").attrib[
-                "name"
-            ]
-            for sm in modifications
-        }
         lookup = {}
-        for pep in self.root.findall(".//{*}Peptide"):
-            id = pep.attrib.get("id", "")
-            lookup[id] = {"modifications": []}
-            lookup[id]["sequence"] = pep.find(".//{*}PeptideSequence").text
-            for child in pep.findall(".//{*}Modification"):
-                lookup[id]["modifications"].append(
-                    f"{modification_mass_map[child.attrib['monoisotopicMassDelta']]}:{child.attrib['location']}"
-                )
-            lookup[id]["modifications"] = ";".join(lookup[id]["modifications"])
+        for pep_sequence, attribs in peptide_lookup.items():
+            lookup[pep_sequence] = {
+                "modifications": [],
+                "sequence": attribs["sequence"],
+            }
+            if len(attribs["modifications"]) != 0:
+                for mod in attribs["modifications"]:
+                    monoisotopicMassDelta = mod["monoisotopicMassDelta"]
+                    location = mod["location"]
+                    lookup[pep_sequence]["modifications"].append(
+                        f"{modification_mass_map[monoisotopicMassDelta]}:{location}"
+                    )
+            lookup[pep_sequence]["modifications"] = ";".join(
+                lookup[pep_sequence]["modifications"]
```

Comment:
might need sorting?

Reply:
yes and no, would be nice but is resorted during clean up anyways